### DISCUSSION TITLE
Deleting a folder in a directory

### DISCUSSION THREAD

**Discussion by BioNCphilic** [earthlab-education/Earth-Analytics-AY24#516] [General]
I somehow managed to create three different paths for my ecoregions download file. I'm trying to delete two of them. Here's what I found as a [suggestion] (https://www.geeksforgeeks.org/delete-a-directory-or-file-using-python/#example-1-delete-all-directories-from-a-directory) and here's my code. 

# Directory name
directory = "resolve_ecoregions)"

# Parent Directory
parent = "D:/earth-analytics/data/species-project"

# Path
path = os.path.join(parent, directory)

# Remove the Directory
# "resolve_ecoregions"
os.rmdir(earth-analytics/data/species-project)

The error message I get is:

NameError                                 
---> [12] os.rmdir(earth-analytics/data/species-project)

NameError: name 'earth' is not defined

If i use os.rmdir(/earth-analytics/data/species-project) the error message I get is: SyntaxError: invalid syntax

**Comment by earthlab-test-student** [comment]
Hi @BioNCphilic , this looks like potentially a quick fix -- you will need to put quotes around the directory path! Python is complaining that there's no `earth` variable defined because your path starts with an _unquoted_ `earth`, followed by a `-` which Python interprets as a subtraction operator.

In general, I do not recommend hard-coding file paths with the separator like this. I know this is a one-off, but if you intend to keep the code in your notebook, use `os.path.join('earth-analytics', 'data', 'species-project')` instead. This will help to ensure that your code is reproducible on both Windows and UNIX systems!

**Reply by eculler** [reply]
Oops, I did this as my test student account. This is Elsa! ^
